node/src/components/fetcher/trie_fetcher.rs

```diff
@@ -97,12 +96,18 @@ where
     fn merge(&mut self, other: PartialChunks<I>) {
         self.chunks.extend(other.chunks);
         self.responders.extend(other.responders);
-        self.peers.extend(other.peers);
+        // this is quadratic in the number of peers, but it's never huge, so it shouldn't be a
```

Comment:
Just FYI, on TestNet we have ~350 peers and ~180 on MainNet.

Reply:
Hmm, I expected about 100 peers, but not 350. Do you think I should optimize that, then?

Reply:
A set _and_ a vec? :confused:

Reply:
Bloom filter for membership test?

Reply:
Is there a read implementation of Bloom filters I could use?
Otherwise, I could just use a set. I was a bit hesitant to do that, because it means additional trait bounds that will have to be added in multiple places, but oh well.

Reply:
So... I'm afraid this is far too complicated for something where we have a simple solution.

We don't need this component and we don't need to track multiple peers in this component.

Reply:
I agree that a Bloom filter would be overkill. I'd just use a set and a vec?